APE17.rst

```diff
@@ -0,0 +1,701 @@
+longer rely on astropy-helpers could make the ``python setup.py build_docs``
+command print out information about how to run the tests, and
+optionally include a link to more information::
+
+    $ python setup.py build_docs
+
+    Note: building the documentation is no longer done using ``python setup.py build_docs``,
+    but instead you will need to run the following command:
+
+        tox -e build_docs
+
+    If you don't already have tox installed, you can install it with:
+
+        pip install tox
+
+    For more information, see https://docs.astropy.org/...
+
+    $
+
+We could then provide a clear documentation page in a centralized place on how
+to use tox to build the docs, or how to use the ``make html`` approach.
+
+These changes will have no impact on ReadTheDocs as that service never made use
+of the ``build_docs`` command, instead invoking sphinx directly with
+``sphinx-build``.
+
+Version helpers
+~~~~~~~~~~~~~~~
+
+Currently, astropy-helpers includes version helpers that take the version
+defined in ``setup.cfg`` or ``setup.py`` and add a developer string when in a
+checked out version of a git repository. The developer string consists of
+``version.devN`` where N is the number of commits since a release.
+
+The Python Packaging Authority (PyPA) now provide a setuptools extension called
+`setuptools_scm`_ which can entirely replace the version helpers in
+astropy-helpers. The way this package works is that versions are no longer
+specified in ``setup.cfg`` or ``setup.py`` - instead the versions are taken from
+tags. Developer version strings produced in this way are much more sophisticated
+and can indicate for example if the working copy is clean or has local changes,
+and whether it is a stable tagged version or a developer version. We note that
+the default way of setting up setuptools_scm results in the version string not
+being updated automatically in ‘editable’ installs of a package (i.e. ``pip
+install -e .``), however `a workaround
+<https://github.com/pypa/setuptools_scm/issues/273>`_ is available.
+
+Switching to `setuptools_scm`_ requires minimal configuration, which is well
+described in its documentation and we therefore do not repeat here. However, one
+subtlety is that since it relies on tags to determine versions, for packages
+such as the core astropy package which have all their tags on branches (at least
+in recent years), we will need to add a 'developer' tag on ``master`` straight
+after branching, e.g., after creating a ``v4.0.x`` branch we should tag the next
+commit on ``master`` as ``v4.1.dev``. In any case, the documentation about `how
+to release the core package
+<http://docs.astropy.org/en/stable/development/releasing.html>`_ as well as
+`packages using the astropy package template
+<http://docs.astropy.org/en/stable/development/astropy-package-template.html#releasing-a-python-package>`_
+will need to be updated to reflect the use of setuptools_scm.
```

Comment:
I think the template needs to be updated for quite a bit more than just `setuptools_scm`, right?  That probably should be stated explicitly as a mandatory part of the process.

Reply:
I guess it's in the "timeline" section, although it's not clear exactly what makes this special, then?

Reply:
This is under the "Version Helpers" section, and is describing things which reference the old version stuff that would need adjusting for the new version stuff.

Reply:
Fixed in https://github.com/astropy/astropy-APEs/pull/52/commits/7695c58e8e539acf8dc1f94e741448a1045c4d7e